execution/executor-types/src/state_checkpoint_output.rs

```diff
@@ -129,4 +129,16 @@ impl StateCheckpointOutput {
             Ok(())
         })
     }
+
+    pub fn ensure_no_discard_or_retry(&self) -> Result<()> {
+        ensure!(
+            self.txns.to_discard.is_empty(),
+            "Syncing discarded transactions"
+        );
+        ensure!(
+            self.txns.to_retry.is_empty(),
+            "Chunk crosses epoch boundary.",
```

Comment:
Should we just say `Syncing retried transactions` here?
(is exceeding block gas limit also a retry?)